packages/phoenix-ng/projects/phoenix-ui-components/lib/components/ui-menu/event-data-explorer/event-data-explorer-dialog/event-data-explorer-dialog.component.ts

```diff
@@ -96,16 +99,41 @@ export class EventDataExplorerDialogComponent {
     this.dialogRef.close();
   }
 
+  private async unzip(data: ArrayBuffer) {
+    const archive = new JSZip();
+    await archive.loadAsync(data);
+    let fileData = '';
+    let multiFile = false;
+    for (const filePath in archive.files) {
+      if (multiFile) {
+        console.error(
+          'Zip archive contains more than one file. Ignoring all but first'
+        );
+        break;
+      }
+      fileData = await archive.file(filePath).async('string');
+      multiFile = true;
+    }
+    return fileData;
+  }
+
   makeRequest(
     urlPath: string,
-    responseType: 'json' | 'text',
+    responseType: 'json' | 'text' | 'blob',
```

Comment:
Instead of passing blob here, we can pass `arrayBuffer` and that will give us the data as the `ArrayBuffer` type. And at call time we can use the `readZipFile` helper function I shared a link to to read the data. This way we can avoid adding any conditions to this function.

I added a snippet for this in the third comment.

Reply:
Not so easy, 'blob' here is used to get the data form the `Response` object. So now I have a line 
```javascript
      .then((res) => res[responseType == 'arrayBuffer' ? 'blob' : responseType]())
```